lib/kennel/file_cache.rb

```diff
@@ -44,11 +44,14 @@ def load_data
     end
 
     def persist
+      return if @data.nil?
```

Comment:
prefer to make load_data fallback to `{}` we should never have nil data here

Reply:
OK, do have a specific patch to propose?

Reply:
I have a _suspicion_ as to how my cache ended up containing simply a `nil` (the cache file was literally 3 bytes long, the exact same 3 bytes that `Marshal.dump(nil)` gives), i.e. in that specific case, it wasn't the non-flushing problem, it was the nil problem. However ... we kinda _already_ fall back to `{}`, right? Except it doesn't seem to work.

Reply:
on line 40 we should say `|| {}`

Reply:
That's kinda missing the point, isn't it? That says: if the cache is `nil` (i.e. already corrupted), then try to un-corrupt it. Shouldn't we be trying to make sure it doesn't corrupt in the first place?